extensions/filesystem_transport/main.md

```diff
@@ -0,0 +1,77 @@
+
+All other properties of DIDComm-style `serviceEndpoint` objects are [standard](https://identity.foundation/didcomm-messaging/spec/#service-endpoint) with this DIDComm transport extension.
+
+#### Format and Naming conventions
+
+The format of messages as files is exactly as given in the [DIDComm v2 spec](https://identity.foundation/didcomm-messaging/spec/). This means files contain exactly and only the bytes that one would expect for a [DIDComm Encrypted Message](https://identity.foundation/didcomm-messaging/spec/#didcomm-encrypted-messages), a [DIDComm Signed Message](https://identity.foundation/didcomm-messaging/spec/#didcomm-signed-messages), or a [DIDComm Plaintext Message](https://identity.foundation/didcomm-messaging/spec/#didcomm-encrypted-messages) if transferred over the wire. Conforming implementations MUST apply the appropriate filename extension as given in the spec (*.dcem, *.dcsm, *.dcpm, respectively). This allows DIDComm messages to be displayed in filesystem GUI tools like Windows Explorer or Mac Finder with appropriate icons, and with correctly inferred [IANA media types](https://identity.foundation/didcomm-messaging/spec/#iana-media-types).
+
+Files in a folder that lack these filename extensions MUST be ignored by conforming implementations. This allows agents to hide messages that have already been processed, or that are still being built, by simply changing their extension.
+
+Aside from the extension, the remainder of the filename of an individual message is arbitrary. However, it MAY include the value from the message's `id` header to guarantee uniqueness.
+
+#### Ordering
+
+When more than one DIDComm message is found in a single folder, receivers MUST process them in a predictable order to simulate the order-of-arrival semantics that are taken for granted in most other transports.
+
+Two ordering semantics are supported. The simplest and default is to process files in order of ascending last modification date. Usually this suffices. However, there may be corner cases where this is inadequate. Therefore, it is also possible for senders to force an arbitrary order by naming files with an __ordinal prefix__ that consists of digits followed by a double hyphen, as in `01--msg.dcem`, `02--msg.dcem`. Leading zeros are optional; the double hyphen is required to distinguish this convention from names based on UUIDs and other identifiers that use digits and hyphens. When filenames of messages begin with an ordinal prefix, messages MUST be processed by a receiver in the numeric (NOT lexicographic) order implied by the prefixes.
```

Comment:
For the second option of ordering, an implementor would like to have a way to avoid the sender the need to scan/read all file names in the folder to check which was the last digit used (assuming the sender does not keep track of them; I'm thinking on a mediator for example). One option is to use the unixtime mills as the ordinal prefix.

Reply:
Excellent suggestion. I'll add a sentence about that.